scipy/spatial/_hausdorff.pyx

```diff
@@ -0,0 +1,67 @@
+"""
+Directed Hausdorff Code
+
+.. versionadded:: 0.19.0
+
+"""
+#
+# Copyright (C)  Tyler Reddy, Richard Gowers, and Max Linke, 2016
+#
+# Distributed under the same BSD license as Scipy.
+#
+
+import numpy as np
+cimport numpy as np
+cimport cython
+from libc.math cimport sqrt
+
+__all__ = ['directed_hausdorff']
+
+@cython.boundscheck(False)
+def directed_hausdorff(double[:,::1] ar1, double[:,::1] ar2, seed=0):
+
+    cdef double cmax, cmin, d
+    cdef int break_occurred
+    cdef int N1 = ar1.shape[0]
+    cdef int N2 = ar2.shape[0]
+    cdef int data_dims = ar1.shape[1]
+    cdef unsigned int i, j, k
+    cdef unsigned int i_store, j_store, i_ret, j_ret = 0
+    cdef long[:] resort1, resort2
+
+    # shuffling the points in each array generally increases the likelihood of
+    # an advantageous break in the inner search loop and never decreases the
+    # performance of the algorithm
+    rng = np.random.RandomState(seed)
+    resort1 = np.arange(N1)
+    resort2 = np.arange(N2)
+    rng.shuffle(resort1)
+    rng.shuffle(resort2)
+    ar1 = np.asarray(ar1)[resort1]
+    ar2 = np.asarray(ar2)[resort2]
+
+    cmax = 0
+    for i in range(N1):
+        break_occurred = 0
+        cmin = np.inf
+        for j in range(N2):
+            d = 0
+	    # faster performance with square of distance
+	    # avoid sqrt until very end
+            for k in range(data_dims):
+                d += (ar1[i, k] - ar2[j, k]) * (ar1[i, k] - ar2[j, k])
+            if d < cmax: # early break
+                break_occurred += 1
+                break
+
+            if d < cmin:
+                cmin = d
+                i_store = i
+                j_store = j
+
+        if cmin > cmax and cmin != np.inf and break_occurred == 0:
```

Comment:
now I see it - the ``break_occurred`` name confused me. this if-statement is only ever true if the break didn't occur. ``break_occurred`` should also be a boolean not an int, since you only ever test for a single break